compiler/rustc_type_ir/src/elaborate.rs

```diff
@@ -111,6 +113,18 @@ impl<I: Interner, O: Elaboratable<I>> Elaborator<I, O> {
             return;
         };
 
+        // PERF(sized-hierarchy): To avoid iterating over sizedness supertraits in
+        // parameter environments, as an optimisation, sizedness supertraits aren't
+        // elaborated, so check if a `Sized` obligation is being elaborated to a
+        // `MetaSized` obligation and emit it. Candidate assembly and confirmation
+        // are modified to check for the `Sized` subtrait when a `MetaSized` obligation
+        // is present.
```

Comment:
```suggestion
        // elaborated. Candidate assembly and confirmation are modified to lazily
        // check for the `Sized` subtrait when a `MetaSized` obligation is present.
```
emit/omit?

Reply:
also, please explicitly match on `Filter` here even if we treat both variants the same for now and add a FIXME that we need to decide this on a per use basis.

Hmm, alternatively add a separate 2 state enum to the `elaborate`
```rust
enum ElaborateSized {
    Yes,
    No,
}
``` 

E.g. the overlap check in coherence uses elaboration and this one should properly elaborate `MetaSized`

Reply:
Fixed, added `ElaborateSized`, defaulting it to `ElaborateSized::No` in the `Elaborator`, and then toggling it to `ElaborateSized::Yes` in coherence.